gui/wxpython/datacatalog/tree.py

```diff
@@ -645,34 +645,40 @@ def OnCopyMap(self, event):
         if len(self.copy_layer) > 1:
             label = _("{c} maps marked for copying.").format(c=len(self.selected_layer))
         else:
-            label = _("Map <{layer}> marked for copying.").format(layer=self.copy_layer[0].label)
+            label = _("Map <{layer}> marked for copying.").format(layer=self.copy_layer[0].data['name'])
         self.showNotification.emit(message=label)
 
     def OnRenameMap(self, event):
         """Rename layer with dialog"""
-        old_name = self.selected_layer[0].label
+        old_name = self.selected_layer[0].data['name']
         gisrc, env = gscript.create_environment(
-            self.selected_grassdb[0].label,
-            self.selected_location[0].label,
-            self.selected_mapset[0].label)
-        new_name = self._getNewMapName(
-            _('New name'),
-            _('Rename map'),
-            old_name,
-            env=env,
-            mapset=self.selected_mapset[0].label,
-            element=self.selected_layer[0].data['type'])
-        if new_name:
-            self.Rename(old_name, new_name)
+            self.selected_grassdb[0].data['name'],
+            self.selected_location[0].data['name'],
+            self.selected_mapset[0].data['name'])
+        try:
+            new_name = self._getNewMapName(
+                _('New name'),
+                _('Rename map'),
+                old_name,
+                env=env,
+                mapset=self.selected_mapset[0].data['name'],
+                element=self.selected_layer[0].data['type'])
+            if new_name:
+                self.Rename(old_name, new_name)
+                self.ReloadTreeItems()
+        except Exception as e:
```

Comment:
Which exception are you trying to catch here? Raised in _getNewMapName or Rename?

Reply:
Remove it or explain why it needs to be there.